edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -95,30 +96,106 @@ impl std::fmt::Debug for DockerModuleRuntime {
+                let config_path_buf = c;
+                let config_path_string_result =
+                    config_path_buf.clone().into_os_string().into_string();
+                let config_path_str;
+                match config_path_string_result {
+                    Ok(c) => config_path_str = c,
+                    Err(_) => config_path_str = "".to_string(),
+                }
+                let mut image_with_tag_parts = image_with_tag.split(':');
+                let gun = image_with_tag_parts
+                    .next()
+                    .expect("split always returns atleast one element")
+                    .to_owned();
+                let gun: Arc<str> = gun.into();
+                let tag = image_with_tag_parts.next().unwrap_or("latest").to_owned();
+                let notary_auth = format!("{}:{}", username, password);
+                let notary_auth = base64::encode(&notary_auth);
+                let lock = self.notary_lock.clone();
+                let mutex = MutexFuture(lock);
+                future::Either::A(
+                    mutex
+                        .and_then({
+                            let gun = gun.clone();
+                            move |lock| {
+                                notary::notary_lookup(
+                                    &notary_auth,
+                                    &gun,
+                                    &tag,
+                                    &config_path_str,
+                                    lock,
+                                )
+                            }
+                        })
+                        .map(move |digest| (format!("{}@{}", gun, digest), true)),
+                )
+            } else {
+                info!("{} is not enabled for notary content trust", hostname);
+                future::Either::B(futures::future::ok((image_with_tag, false)))
+            }
+        } else {
+            future::Either::B(futures::future::ok((image_with_tag, false)))
+        };
 
-        let response = creds
-            .map(|creds| {
-                self.client
+        let creds = config.auth().cloned();
+        let client_copy = self.client.clone();
+        let response = image
+            .and_then(|(image, is_content_trust_enabled)| {
+                let creds = match creds {
+                    Some(a) => {
+                        let json = serde_json::to_string(&a).with_context(|_| {
+                            ErrorKind::RegistryOperation(RegistryOperation::PullImage(
+                                image.clone(),
+                            ))
+                        })?;
+                        base64::encode_config(&json, base64::URL_SAFE)
+                    }
+                    None => String::new(),
+                };
+                Ok((image, is_content_trust_enabled, creds))
+            })
+            .and_then(move |(image, is_content_trust_enabled, creds)| {
+                if is_content_trust_enabled {
```

Comment:
These logging can be in the upper section. Am I right?

Reply:
Ya but the intention was to enable logging right before image creation